src/firmware/app/world/firmware_robot.c

```diff
@@ -3,6 +3,41 @@
 #include <math.h>
 #include <stdlib.h>
 
+// these are set to decouple the 3 axis from each other
+// the idea is to clamp the maximum velocity and acceleration
+// so that the axes would never have to compete for resources
+#define TIME_HORIZON 0.05f  // s
+
+typedef void (*TrajectoryFollower_t)(const FirmwareRobot_t*, PositionTrajectory_t, size_t,
+                                     float);
+typedef void (*ApplyDirectPerWheelPower_t)(
+    const FirmwareRobot_t*, TbotsProto_DirectControlPrimitive_DirectPerWheelControl);
+typedef void (*SetLocalVelocity_t)(
+    const FirmwareRobot_t*, TbotsProto_DirectControlPrimitive_DirectVelocityControl);
+typedef void (*StopRobot_t)(const FirmwareRobot_t*, TbotsProto_StopPrimitive_StopType);
+
+/**
+ * Determines the rotation acceleration after setup_bot has been used and
+ * plan_move has been done along the minor axis. The minor time from bangbang
+ * is used to determine the rotation time, and thus the rotation velocity and
+ * acceleration. The rotational acceleration is clamped under the MAX_T_A.
+ *
+ * @param pb [in/out] The PhysBot data container that should have minor axis time and
+ * will store the rotational information
+ * @param avel The rotational velocity of the bot
+ */
+void plan_move_rotation(PhysBot* pb, float avel);
+
+void plan_move_rotation(PhysBot* pb, float avel)
+{
+    pb->rot.time = (pb->min.time > TIME_HORIZON) ? pb->min.time : TIME_HORIZON;
+    // 1.4f is a magic constant to force the robot to rotate faster to its final
+    // orientation.
+    pb->rot.vel   = 1.4f * pb->rot.disp / pb->rot.time;
+    pb->rot.accel = (pb->rot.vel - avel) / TIME_HORIZON;
+    limit(&pb->rot.accel, MAX_T_A);
+}
+
```

Comment:
can this live in `physbot.h/c`?

Reply:
IIRC, `plan_move_rotation` was declared and used only in `physbot.c` so it seems like it was only intended as a private method. Open to moving it elsewhere though.